src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arrayWithYears = people
+    .filter((person) => {
+      return century ? person.sex === 'm'
+      && Math.ceil(person.died / 100) === century : person.sex === 'm';
+    })
+    .map((person) => {
+      return person.died - person.born;
+    });
+
+  const yearsAmount = arrayWithYears.reduce((acc, years) => acc + years, 0);
+
+  return yearsAmount / arrayWithYears.length;
 }
```

Comment:
u do same logic (calc avarage age) several times, lets create helper function for this